cmake: sysbuild: Fix downgrade  prevention UICR provisioning (#29864)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  Use their PSA Crypto counterparts instead.

> * Fixed issue whereby UICR would not get provisioned with structures for monotonic counters, for :kconfig:option`SB_CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION`, when MCUboot was the sole bootloader and Partition Manager has been disabled.

Reviewer: ```suggestion
* Fixed an issue where UICR was not provisioned with monotonic counter structures, when :kconfig:option`SB_CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION` was enabled, MCUboot was the only bootloader, and Partition Manager was disabled.
```
Please check the meaning.

> * Fixed issue whereby UICR would not get provisioned with structures for monotonic counters, for :kconfig:option`SB_CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION`, when MCUboot was the sole bootloader and Partition Manager has been disabled.

Reviewer: Seems like this entry will go in the 3.4 release notes. If so, please remove the entry from this PR: https://github.com/nrfconnect/sdk-nrf/pull/29865/changes#diff-9460bd2e76b8fe433e47a738942f77d38f18d5967374101c756a015372b8cc22

Author: I will split the commit to code and doc.

Reviewer: My bad, I meant to remove the entry from this PR 29864. You could add a comment in PR https://github.com/nrfconnect/sdk-nrf/pull/29095 to have this entry in the 3.4 release notes.
